iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -188,47 +159,26 @@ TEST_F(ServiceDiscovery_test, OfferExistingServiceMultipleTimesIsRedundant)
     publisher.offer();
     this->InterOpWait();
 
-    auto serviceContainer = sut.findService(SERVICE_DESCRIPTION.getServiceIDString(),
-                                            SERVICE_DESCRIPTION.getInstanceIDString(),
-                                            SERVICE_DESCRIPTION.getEventIDString());
+    findService(SERVICE_DESCRIPTION.getServiceIDString(),
+                SERVICE_DESCRIPTION.getInstanceIDString(),
+                SERVICE_DESCRIPTION.getEventIDString());
 
     ASSERT_THAT(serviceContainer.size(), Eq(1U));
     EXPECT_THAT(*serviceContainer.begin(), Eq(SERVICE_DESCRIPTION));
-
-    sut.findService(SERVICE_DESCRIPTION.getServiceIDString(),
-                    SERVICE_DESCRIPTION.getInstanceIDString(),
-                    SERVICE_DESCRIPTION.getEventIDString(),
-                    findHandler);
-    compareAndClearServiceContainers(serviceContainer, searchResultOfFindServiceWithFindHandler);
 }
 
+
 TEST_F(ServiceDiscovery_test, FindSameServiceMultipleTimesReturnsSingleInstance)
 {
     ::testing::Test::RecordProperty("TEST_ID", "21948bcf-fe7e-44b4-b93b-f46303e3e050");
     const iox::capro::ServiceDescription SERVICE_DESCRIPTION("service", "instance", "event");
     iox::popo::UntypedPublisher publisher(SERVICE_DESCRIPTION);
 
-    auto serviceContainer = sut.findService(SERVICE_DESCRIPTION.getServiceIDString(),
-                                            SERVICE_DESCRIPTION.getInstanceIDString(),
-                                            SERVICE_DESCRIPTION.getEventIDString());
-    ASSERT_THAT(serviceContainer.size(), Eq(1U));
-    EXPECT_THAT(*serviceContainer.begin(), Eq(SERVICE_DESCRIPTION));
-    sut.findService(SERVICE_DESCRIPTION.getServiceIDString(),
-                    SERVICE_DESCRIPTION.getInstanceIDString(),
-                    SERVICE_DESCRIPTION.getEventIDString(),
-                    findHandler);
-    compareAndClearServiceContainers(serviceContainer, searchResultOfFindServiceWithFindHandler);
-
-    serviceContainer = sut.findService(SERVICE_DESCRIPTION.getServiceIDString(),
-                                       SERVICE_DESCRIPTION.getInstanceIDString(),
-                                       SERVICE_DESCRIPTION.getEventIDString());
+    findService(SERVICE_DESCRIPTION.getServiceIDString(),
+                SERVICE_DESCRIPTION.getInstanceIDString(),
+                SERVICE_DESCRIPTION.getEventIDString());
     ASSERT_THAT(serviceContainer.size(), Eq(1U));
     EXPECT_THAT(*serviceContainer.begin(), Eq(SERVICE_DESCRIPTION));
```

Comment:
Please add again an additional `findService` call to fulfill the test intention.